vm/src/stdlib/builtins.rs

```diff
@@ -338,7 +351,13 @@ mod builtins {
     }
 
     #[pyfunction]
-    fn hasattr(obj: PyObjectRef, attr: PyStrRef, vm: &VirtualMachine) -> PyResult<bool> {
+    fn hasattr(obj: PyObjectRef, attr: PyObjectRef, vm: &VirtualMachine) -> PyResult<bool> {
+        let attr = PyStrRef::try_from_object(vm, attr.clone()).map_err(|_e| {
```

Comment:
```suggestion
        let attr = PyStrRef::try_from_object(vm, attr).map_err(|_e| {
```
attr can be moved

Reply:
error occurs when .clone() is removed.

Reply:
Oh, i am sorry. It is used again for type error below